<a href="https://colab.research.google.com/github/katheforero3/Curso_Estadistica/blob/main/Ejercicio_final.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [25]:
#Cargar datos
from google.colab import files
uploaded = files.upload()

Saving Base Ventas Celulares.csv to Base Ventas Celulares (1).csv


In [26]:
#Importar Librerias
import pandas as pd
import numpy as np
import math
import matplotlib.pyplot as plt

In [27]:
#Cargar el csv
df = pd.read_csv('Base Ventas Celulares.csv')

#Comprobar que los datos fueran correctamente cargados
print(df.head())

   Fecha Codigo         País Distribuidor Producto  Operador  Cantidad  \
0  40181   A640  El Salvador       Ripley  Samsung  Movistar        15   
1  40183   A687       España    Plaza Vea       LG     Entel        29   
2  40185  A1162     Paraguay        Metro    Nokia     Entel        21   
3  40189   A463         Cuba      Oeschle    Apple     Entel        30   
4  40190   A847    Guatemala        Metro       LG     Claro         7   

  Forma de pago   Precio     Total  
0      Efectivo  1750000  26250000  
1      Efectivo  1100000  31900000  
2      Efectivo   950000  19950000  
3      Efectivo   710000  21300000  
4      Efectivo  1610000  11270000  


**Medidas de tendencia central de las columnas Cantidad y Total**

**Calcular media, mediana y moda de Cantidad**

In [28]:
#Calcular la media de la comumna Cantidad
media_cant = df['Cantidad'].mean()
print(f'La media de la cantidad es: {media_cant}')

La media de la cantidad es: 15.587725150100066


In [29]:
#Calcular la mediana de la comumna Cantidad
mediana_cant = df['Cantidad'].median()
print(f'La mediana de la cantidad es: {mediana_cant}')

La mediana de la cantidad es: 16.0


In [30]:
#Calcular la moda de la comumna Cantidad
moda_cant = df['Cantidad'].mode()
print(f'La moda de la cantidad es: {moda_cant}')

La moda de la cantidad es: 0    24
Name: Cantidad, dtype: int64


**Calcular media, mediana y moda de Total**

In [31]:
#Calcular la media de la comumna Total
media_total = df['Total'].mean()
print(f'La media de total es: {media_total}')

La media de total es: 20958672.448298864


In [32]:
#Calcular la mediana de la comumna Total
mediana_total = df['Total'].median()
print(f'La mediana de total es: {mediana_total}')

La mediana de total es: 19400000.0


In [33]:
#Calcular la moda de la comumna Total
moda_total = df['Total'].mode()
print(f'La moda de total es: {moda_total}')

La moda de total es: 0    17100000
Name: Total, dtype: int64


**¿Qué podemos concluir de cada uno de estos datos?**

**Cantidad**

* La media indica que en promedio, se vendieron aproximadamente 15.5877 unidades por operación.

* La mediana, ligeramente mayor que la media, sugiere que la distribución es algo sesgada, probablemente por ventas de pocas unidades en algunos casos.

* La moda (24 unidades) refleja que este número de ventas fue el más frecuente.

**Total**

* La media muestra que el promedio del total de ventas por operación fue de 20958672.448298864 unidades monetarias.

* La mediana es menor que la media, lo que sugiere que hay operaciones con valores muy altos (outliers) que elevan la media.

* La moda (17100000) es el total de ventas que más se repitió, lo cual podría ser típico de transacciones regulares.

**Medidas de dispersión**

**Calcular varianza, desviación estandar y coeficiente de variación de la columna Cantidad**

In [34]:
#Calcular la varianza de la columna Cantidad
varianza_cant = df['Cantidad'].var()
print(f'La varianza de la cantidad es: {varianza_cant}')

La varianza de la cantidad es: 76.13966231158967


In [35]:
#Desviación estandar de la columna Cantidad
desv_est_cant =df['Cantidad'].std()
desv_est_raiz_cant = np.sqrt(varianza_cant)
print(f'La desviación estandar de la cantidad es: {desv_est_cant}')
print(f'La desviación estandar con la raiz de la cantidad es: {desv_est_raiz_cant}')

La desviación estandar de la cantidad es: 8.725804393383436
La desviación estandar con la raiz de la cantidad es: 8.725804393383436


In [36]:
#Coeficiente de variación de la columna Cantidad
cv_cant =(desv_est_cant/media_cant)*100
print(f'El coeficiente de variacion de la cantidad es: {cv_cant:.2f}%')

El coeficiente de variacion de la cantidad es: 55.98%


**Calcular varianza, desviación estandar y coeficiente de variación de la columna Total**

In [37]:
#Calcular la varianza de la columna Total
varianza_total = df['Total'].var()
print(f'La varianza de total es: {varianza_total}')

La varianza de total es: 181587901774480.44


In [38]:
#Desviación estandar de la columna Total
desv_est_total =df['Total'].std()
desv_est_raiz_total = np.sqrt(varianza_total)
print(f'La desviación estandar de total es: {desv_est_total}')
print(f'La desviación estandar con la raiz de total es: {desv_est_raiz_total}')

La desviación estandar de total es: 13475455.531241994
La desviación estandar con la raiz de total es: 13475455.531241994


In [39]:
#Coeficiente de variación de la columna Total
cv_total =(desv_est_total/media_total)*100
print(f'El coeficiente de variacion de total es: {cv_total:.2f}%')

El coeficiente de variacion de total es: 64.30%


**¿Qué podemos concluir de cada uno de estos datos?**

*  Varianza: Es una medida de dispersión absoluta que refleja la variabilidad de los datos. La alta varianza en Total indica una gran dispersión en las ventas totales, mientras que la varianza en Cantidad es más baja, lo que implica que la dispersión en la cantidad vendida es menor.

* Desviación estándar: Proporciona una idea más tangible de la variabilidad en las mismas unidades de las columnas. Las cifras sugieren que las ventas totales tienen una variabilidad mucho mayor que las cantidades vendidas.

* Coeficiente de variación (CV): Permite comparar la dispersión relativa entre diferentes conjuntos de datos. El CV de Total es mayor que el de Cantidad, lo que indica que las ventas totales son más variables en relación con su media, en comparación con la cantidad de productos vendidos.


